port/unix/omrsysinfo.c

```diff
@@ -3722,24 +3744,44 @@ isRunningInContainer(struct OMRPortLibrary *portLibrary, BOOLEAN *inContainer)
 		}
 
 		while (0 == feof(cgroupFile)) {
+			char buffer[PATH_MAX + 1024];
 			char cgroup[PATH_MAX];
 			char subsystems[1024];
 			int32_t hierId = -1;
 
-			rc = fscanf(cgroupFile, PROC_PID_CGROUP_ENTRY_FORMAT, &hierId, subsystems, cgroup);
+			fgets(buffer, PATH_MAX + 1024, cgroupFile);
+			if(0 != ferror(cgroupFile)){
+				int32_t osErrCode = errno;
+				Trc_PRT_isRunningInContainer_fgets_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "fgets failed to read %s file stream with errno=%d", OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				goto _end;
+			} else if (NULL == buffer) {
+				break;
+			}
+			rc = sscanf(buffer, PROC_PID_CGROUP_ENTRY_FORMAT, &hierId, subsystems, cgroup);
 			/* Ensure we didn't overflow */
 			Assert_PRT_true(strlen(subsystems) < 1024);
-			Assert_PRT_true(strlen(cgroup) < PATH_MAX);
+			Assert_PRT_true(strlen(cgroup) < PATH_MAX);		
 
 			if (EOF == rc) {
 				break;
+			} else if (1 == rc) {
+				rc = sscanf(buffer, PROC_PID_CGROUP_SYSTEMD_ENTRY_FORMAT, &hierId, cgroup);
+				Assert_PRT_true(strlen(cgroup) < PATH_MAX); 
+
+				if (2 != rc) {
+					Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
+					rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
+					goto _end;
+				}
+				subsystems[0] = '\0';
 			} else if (3 != rc) {
 				Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
 				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
 				goto _end;
 			}
 
-			if (0 != strcmp(ROOT_CGROUP, cgroup)) {
+			if (0 != strcmp(ROOT_CGROUP, cgroup) && 0 != strcmp(SYSTEMD_INIT_CGROUP, cgroup)) {
```

Comment:
if there is more than one complex conditional statement each statement needs to be wrapped in brackets.  This line should be:
```
if ((0 != strcmp(ROOT_CGROUP, cgroup)) && (0 != strcmp(SYSTEMD_INIT_CGROUP, cgroup))) {
```